macros/contexts/contextUnits.pl

```diff
@@ -0,0 +1,2056 @@
+    );
+
+    Compute("3 apples") == Compute("3 oranges"); # returns 1
+
+will consider apple and oranges as the same unit (both are the
+fundamental unit of C<fruit>).
+
+Finally,
+
+    Context("Units")->addUnits(
+      apples => { fruit => 1, aliases => ["apple"] },
+      oranges => { fruit => 1, aliases => ["orange"], factor => 2 }
+    );
+
+    Compute("1 apple") == Compute("2 oranges"); # returns 1
+
+will make an apple equivalent to two oranges by making both C<apples>
+and C<oranges> be examples of the fundamental unit C<fruit>.
+
+You can remove individual units from the context using the
+C<removeUnits()> method of the context.  For example
+
+    Context("Units")->withUnitsFor("length")->removeUnits("ft", "inch", "mile", "furlong");
+
+removes the English units and their aliases, leaving only the metric
+units.  To remove a unit without removing its aliases, use C<removeUnitsNotAliases()>
+instead.
+
+Note that the units are stored in the context as constants, so to list
+all the units, together with other constants, use
+
+    Context()->constants->names;
+
+The constants that are units have the C<isUnit> property set.  So
+
+    grep {Context()->constants->get($_)->{isUnit}} (Context()->constants->names);
+
+will get the list of units.
+
+
+=head2 Adding units to other contexts
+
+The C<Units> and C<LimitedUnits> contexts are based on the C<Numeric>
+and C<LimitedNumeric> contexts.  You can add units to other contexts
+using the C<context::Units::extends()> function.  For example,
+
+    Context(context::Units::extending("Fraction")->withUnitsFor("length"));
+
+would allow you to use fractions with units.
+
+In addition to the name of the context to extend, you can pass options
+to C<context::Units::extending()>, as in
+
+    $context = Context(context::Units::extending("LimitedFraction", limited => 1));
+    $context->addUnitsFor("length");
+
+In this case, the C<limited => 1> option indicates that no operations
+are allowed between numbers with units, and since the
+C<LimitedFraction> context doesn't allow operations otherwise, you
+will only be able to enter fractions or whole numbers, with or without
+units, or a unit without a number.
+
+The available options and their defaults are
+
+    keepNegativePowers => 1,     Preserve use of negative powers so C<m s^-1> will
+                                 not be shown as C<m/s> (but will still match it).
+    useNegativePowers => 0       Always use negative powers instead of fractions?
+    limited => 0                 Don't allow operations on numbers with units.
+    exactUnits => 0              Require student units to exactly match correct ones
+                                   in both order and use of negative powers
+    sameUnits => 0               Require student units to match correct ones
+                                   not scaled versions
+    partialCredit => .5          Partial credit if answer is right but units
```

Comment:
I think you imply that 'units are incorrect'.

Reply:
Fixed.